src/planscape/plan/tests.py

```diff
@@ -1255,6 +1255,83 @@ def test_create_scenario_with_weights(self):
             scenario=scenario.pk)
         self.assertEqual(weighted_pris.count(), 1)
 
+class UpdateScenarioTest(TransactionTestCase):
+    def setUp(self):
+        self.geometry = {'type': 'MultiPolygon',
+                         'coordinates': [[[[1, 2], [2, 3], [3, 4], [1, 2]]]]}
+        stored_geometry = GEOSGeometry(json.dumps(self.geometry))
+
+        self.base_condition = BaseCondition.objects.create(
+            condition_name="cond1", condition_level=ConditionLevel.ELEMENT)
+        self.condition1 = Condition.objects.create(
+            condition_dataset=self.base_condition, raster_name="name1", is_raw=False)
+
+        self.user = User.objects.create(username='testuser')
+        self.user.set_password('12345')
+        self.user.save()
+
+        self.plan = create_plan(
+            self.user, 'plan', stored_geometry, [])
+        self.project = Project.objects.create(
+            owner=self.user, plan=self.plan, max_budget=100)
+        self.project_area = ProjectArea.objects.create(
+            owner=self.user, project=self.project,
+            project_area=stored_geometry, estimated_area_treated=200)
+        self.scenario = Scenario.objects.create(
+            owner=self.user, plan=self.plan, project=self.project, notes='old note')
+        self.weight = ScenarioWeightedPriority.objects.create(
+            scenario=self.scenario, priority=self.condition1, weight=2)
+
+    def test_create_scenario_invalid_id(self):
+        self.client.force_login(self.user)
+        response = self.client.patch(
+            reverse('plan:update_scenario'),
+            {'id': 10},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_update_not_owned_scenario(self):
+        response = self.client.patch(
+            reverse('plan:update_scenario'),
+            {'id': self.scenario.pk},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_update_note(self):
+        self.client.force_login(self.user)
+        response = self.client.patch(
+            reverse('plan:update_scenario'),
+            {'id': self.scenario.pk, 'notes': "new note"},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 200)
+        scenario = Scenario.objects.get(id=self.scenario.pk)
+        self.assertEqual(scenario.project, self.project)
+        self.assertEqual(scenario.notes, "new note")
+        self.assertEqual(scenario.status, Scenario.ScenarioStatus.INITIALIZED)
+
+    def test_remove_note(self):
+        self.client.force_login(self.user)
+        response = self.client.patch(
+            reverse('plan:update_scenario'),
+            {'id': self.scenario.pk, 'notes': None},
```

Comment:
What does a url with {'notes': None} look like?

Is it when "notes" is missing as a url parameter?
Or is it when the url has "&notes=None", or "&notes=&x=y"?

Double-checking endpoint syntax because we don't want to accidentally erase notes that should be saved.

(and/or maybe I'm just ignorant about the FE request syntax?)

Reply:
good question. @lattemilk probably knows better than i do!

Reply:
I think the url would be the latter where notes has an empty value: "&notes=&x=y". :P correct me if I'm wrong though.

Reply:
I see!
If we're unsure, is there a way to reconfigure the test to take the url into account?
Because I can also see "&notes=&x=y" resulting in {'notes': ''}.

Reply:
I think the FE would just not include the notes field in the update request if the user doesn't want to update it, assuming the endpoint only updates fields that have been sent with the patch request.

Reply:
verified that the http request can be made with a data field like this:
{'id': [self.scenario.pk](http://self.scenario.pk/), 'notes': None}

thanks for checking on this!